lib/thread_local.rb

```diff
@@ -101,17 +101,18 @@ def forget_vanished_threads!
     end
   end
 
-  def define_initializer!
-    ObjectSpace.define_finalizer self do
```

Comment:
Exactly. That's what I was trying to solve with this commit, but per your other comment, I bet you're right - I bet `thread_local` is still being captured because it's in scope, even though it's not actually used by the finalizer body.